crates/bevy_winit/src/winit_windows.rs

```diff
@@ -17,18 +17,39 @@ impl WinitWindows {
         #[cfg(target_os = "windows")]
         let winit_window_builder = {
             use winit::platform::windows::WindowBuilderExtWindows;
-            winit::window::WindowBuilder::new()
-                .with_drag_and_drop(false)
+            winit::window::WindowBuilder::new().with_drag_and_drop(false)
         };
 
         #[cfg(not(target_os = "windows"))]
-        let winit_window_builder = winit::window::WindowBuilder::new();
+        let mut winit_window_builder = winit::window::WindowBuilder::new();
 
-        let winit_window = winit_window_builder
-            .with_title(&window.title)
-            .with_inner_size(winit::dpi::PhysicalSize::new(window.width, window.height))
-            .build(&event_loop)
-            .unwrap();
+        winit_window_builder = winit_window_builder.with_title(&window.title);
+
+        let winit_window = match window.mode {
+            WindowMode::BorderlessFullscreen => {
+                let winit_window = winit_window_builder.build(&event_loop).unwrap();
+                winit_window.set_fullscreen(Some(winit::window::Fullscreen::Borderless(
+                    winit_window.current_monitor(),
+                )));
+                winit_window
+            }
+            WindowMode::Fullscreen { use_size } => {
+                let winit_window = winit_window_builder.build(&event_loop).unwrap();
+
+                winit_window.set_fullscreen(Some(winit::window::Fullscreen::Exclusive(
+                    match use_size {
+                        true => get_fitting_videomode(&winit_window.current_monitor(), &window),
+                        false => get_best_videomode(&winit_window.current_monitor()),
+                    },
+                )));
+                winit_window
+            }
+            _ => winit_window_builder
+                .with_inner_size(winit::dpi::PhysicalSize::new(window.width, window.height))
+                .with_resizable(window.resizable)
+                .build(&event_loop)
+                .unwrap(),
+        };
 
```

Comment:
All of the common settings can be moved back down here after the match if it returns a builder.

```rust
        let winit_window = winit_window_builder
            .with_title(&window.title)
            .build(&event_loop)
            .unwrap();
```